extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,19 +1,148 @@
 import {Services} from '#service';
 
+import {user} from '#utils/log';
+
 import {getElementConfig} from 'extensions/amp-story/1.0/request-utils';
 
+import {assertHttpsUrl} from '../../../src/url';
 import {
   Action,
   ShoppingConfigDataDef,
 } from '../../amp-story/1.0/amp-story-store-service';
 
+/** @const {!Object<!Object<string, !Array<function>>>} */
+
+const aggregateRatingValidation = {
+  'ratingValue': [validateRequired, validateNumber],
+  'reviewCount': [validateRequired, validateNumber],
+  'reviewUrl': [validateRequired, validateURLs],
+};
+
+export const productValidationConfig = {
+  /* Required Attrs */
+  'productUrl': [validateRequired, validateString],
+  'productId': [validateRequired, validateString],
+  'productTitle': [validateRequired, validateString],
+  'productBrand': [validateRequired, validateString],
+  'productPrice': [validateRequired, validateNumber],
+  'productImages': [validateRequired, validateURLs],
```

Comment:
'productImages' is an array of objects with `url` and `alt` fields that need to be validated.

Reply:
As per Caleb's comment in https://github.com/ampproject/amphtml/pull/37474#discussion_r816292932 ,
I have included Array of Object validation for `productImages` in `createValidateConfigArray`